datatables.go

```diff
@@ -482,9 +482,23 @@ func (p *Parser) bindWeapon(entity *st.Entity, wepType common.EquipmentElement)
 	var currentOwner *common.Player
 	if wep, ok := p.gameState.weapons[entityID]; ok {
 		currentOwner = wep.Owner
+
+		if p.gameState.weapons[entityID].UniqueID() == 0 {
+			// If we are here, we already have a player that holds this weapon
+			// so the zero-valued Equipment instance was already created in bindPlayer()
+			// In this case we should create a new Equipment instance with non-zero unique id
+			// but having the same memory address so player's rawWeapons would still have a pointer to it
+			*wep = *(common.NewEquipment(wepType))
+		} else {
+			// Something horrible has happened
+			p.eventDispatcher.Dispatch(events.ParserWarn{
+				Message: "trying to bind a new weapon but its entity is already bound",
+			})
+		}
```

Comment:
I've wanted to start using `unassert` a bit more over doing warnings that never happen.
This leads to better code coverage and we actually notice if an assertion fails, since `ParserWarn` is ignored.

assertions are disabled in release builds so this shouldn't interfere with anyone else's code.

I think the new way to go should be:
- if the regression set never goes down some path use `unassert` instead
- otherwise `ParserWarn` *might* be suitable

```suggestion
		unassert.True(p.gameState.weapons[entityID].UniqueID() != 	0)
		// If we are here, we already have a player that holds this weapon
		// so the zero-valued Equipment instance was already created in bindPlayer()
		// In this case we should create a new Equipment instance with non-zero unique id
		// but having the same memory address so player's rawWeapons would still have a pointer to it
		*wep = *(common.NewEquipment(wepType))
```

NOTE: I should probably add `unassert.NotSame()` which would be cleaner

Reply:
PS: we should have probably used `unassert` in the last PR as well, but I forgot about it